core/src/test/java/org/apache/calcite/test/enumerable/AbstractEnumerableTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test.enumerable;
+
+import org.apache.calcite.adapter.java.ReflectiveSchema;
+import org.apache.calcite.config.CalciteConnectionProperty;
+import org.apache.calcite.config.Lex;
+import org.apache.calcite.test.CalciteAssert;
+
+/**
+ * Abstract base class for other tests for enumerable operators.
+ */
+public class AbstractEnumerableTest {
```

Comment:
I am also in favor of the DRY principle but using inheritance to achieve this is questionable (see for instance Item 16, 17 from Effective Java). Often there are better alternatives. In this case almost every caller of this `tester` method passes an HrSchema so you could avoid inheritance by adding a new configuration and use directly `CalciteAssert.AssertThat#with(Config)`.

Reply:
`Test` naming would confuse test framework (it would try to find test methods here) as well.

Alternative way to avoid inheritance is to use interface, and use `default` method for `tester`.

Reply:
Second thinking on this change, I tend to keep it as before. 
Different tests always wants to customize `AssertThat` and I'm hesitate to make a default method/config to offer the testing context

Reply:
I would gentle ping @zabetak @vlsi  to take another look :)